index.js

```diff
@@ -43,6 +49,7 @@ var defer = typeof setImmediate === 'function'
  */
 
 function onFinished (msg, listener) {
+  listener = preserveAsyncContext('OnFinishedListener', listener)
```

Comment:
Should this have something in it to namespace uit to this module? My understanding is that this type name is global in scope and accessible to the user by the way of the init callback in async_hooks.createHook . Please correct me if wrong. If that is the case, should we also then document this name and how to hook it in the docs?

Reply:
You did ask about this, just not in this PR :slightly_smiling_face: -> https://github.com/stream-utils/raw-body/pull/71#discussion_r505166436 (btw, that one is waiting for change requests or approval too).

As for the question: you are correct, that's why I prefixed the name with the module's name (`OnFinished`), but admittedly, this results in the name that a user could come up with for their own hook (`OnFinishedListener`). Maybe `OnFinishedModuleListener` could be a safer alternative?
I agree that documenting it is a good idea, I'll do it as soon as we agree on the name :+1:

Reply:
Ah. So this module's name has a dash in it, which is why i didn't think it was prefixed. If you are going to prefix with the module's name, shouldn't it match?

Reply:
I took a look at existing names and they seem to either be uppercase or upper camel case (https://nodejs.org/api/async_hooks.html#async_hooks_type) so I thought `on-finished` will be `OnFinished` after case conversion. But I have no problem with, say, `on-finished-listener` or `on-finished.listener` if you prefer dashes here :slightly_smiling_face:

Reply:
Well, it's not that I necessarily prefer dashes, but just it seems like it should be prefixed in some way that relates to the module's name. I commented over on the `raw-body` PR there as well, as there is also a `rawbody` module. I assume there could one day also be a `onfinished` module, so it would make sense to probably have the actual module's name as the prefix is what I'm thinking.

Reply:
So your question there had me curious in general, so I looked at the Node.js docs to see if they offer any guidance and the only thing I could find in there was the following really generic statement:

> It is possible to have type name collisions. Embedders are encouraged to use unique prefixes, such as the npm package name, to prevent collisions when listening to the hooks.

So since that is not too prescriptive, I thought about it and realized there must be some other npm modules doing this by now. I tried to use the GitHub search, but there are so many tests coming back. I waded through lots of pages and this was the only other module I found so far: https://github.com/fastify/fastify-request-context/blob/b257465353aec0401a4c427c87df92ac4f0412f1/lib/requestContextPlugin.js#L19

Reply:
That's interesting, but if I were the maintainer of fastify-request-context I wouldn't name an async resource with the exact same name as the module, because how am I going to name another async resource within the same module if there's ever a need to introduce it? Even if it was unlikely that I'll ever need to introduce another one, I'd rather start with `fastify-request-context-store` or something like that.